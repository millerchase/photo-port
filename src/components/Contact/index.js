import React, { useState } from 'react';
import { validateEmail } from '../../utils/helpers';

function ContactForm() {
  const [formState, setFormState] = useState({
    name: '',
    email: '',
    message: ''
  });
  const { name, email, message } = formState;

  const [errorMessage, setErrorMessage] = useState('');

  function handleChange(event) {
    if (event.target.name === 'email') {
      const isValid = validateEmail(event.target.value);

      if (!isValid) {
        setErrorMessage('Your email is invalid');
      } else {
        setErrorMessage('');
      }
    } else {
      if (!event.target.value.length) {
        setErrorMessage(`${event.target.name} is required`);
      } else {
        setErrorMessage('');
      }
    }

    if (!errorMessage) {
      setFormState({ ...formState, [event.target.name]: event.target.value });
    }
  }

  function handleSubmit(event) {
    event.preventDefault();
    console.log(formState);
  }

  return (
    <section>
      <h1>Contact me</h1>
      <form id="contact-form" onSubmit={handleSubmit}>
        <div>
          <label for="name">Name:</label>
          <input
            type="text"
            placeholder="Name"
            defaultValue={name}
            onBlur={handleChange}
            name="name"
          />
        </div>

        <div>
          <label for="email">Email address:</label>
          <input
            type="text"
            placeholder="Email"
            defaultValue={email}
            onBlur={handleChange}
            name="email"
          />
        </div>

        <div>
          <label for="message">Message:</label>
          <textarea
            defaultValue={message}
            name="message"
            onBlur={handleChange}
            rows="5"
          />
        </div>
        {errorMessage && (
          <div>
            <p className="error-text">{errorMessage}</p>
          </div>
        )}
        <button type="submit">Submit</button>
      </form>
    </section>
  );
}

export default ContactForm;
