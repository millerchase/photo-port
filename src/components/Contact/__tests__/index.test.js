import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from '..';

afterAll(cleanup);

describe('Contact component', () => {
  it('renders', () => {
    render(<Contact />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Contact />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders h1 tag', () => {
    const { getByTestId } = render(<Contact />);
    expect(getByTestId('h1tag')).toHaveTextContent('Contact');
  });

  it('renders submit button', () => {
    const { getByTestId } = render(<Contact />);
    expect(getByTestId('submit-btn')).toHaveTextContent('Submit');
  });
});
