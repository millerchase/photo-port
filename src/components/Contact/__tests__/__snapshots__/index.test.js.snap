// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="h1tag"
    >
      Contact me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email address:
        </label>
        <input
          name="email"
          placeholder="Email"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        data-testid="submit-btn"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
